circleci: failed tests (13514): TestSSLEnforcement

The following test appears to have failed:

[#13514](https://circleci.com/gh/cockroachdb/cockroach/13514):

```
E160229 18:33:56.472490 server.go:2159  http: TLS handshake error from 127.0.0.1:45863: tls: first record does not look like a TLS handshake
E160229 18:33:56.546086 server.go:2159  http: TLS handshake error from 127.0.0.1:45864: tls: first record does not look like a TLS handshake
E160229 18:33:56.592202 server.go:2159  http: TLS handshake error from 127.0.0.1:45865: tls: first record does not look like a TLS handshake
E160229 18:33:56.621310 server.go:2159  http: TLS handshake error from 127.0.0.1:45866: tls: first record does not look like a TLS handshake
I160229 18:33:56.666607 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestSSLEnforcement (1.03s)
	authentication_test.go:148: [28]: expected success=true, got err=Post https://127.0.0.1:59208/sql/Execute: stream error: stream ID 13; STREAM_CLOSED
=== RUN   TestParseInitNodeAttributes
I160229 18:33:56.696545 server/context.go:141  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory)
I160229 18:33:56.697142 server/context.go:229  1 storage engine initialized
I160229 18:33:56.697238 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestParseInitNodeAttributes (0.00s)
=== RUN   TestParseJoinUsingAddrs
I160229 18:33:56.699904 server/context.go:141  can't read available memory from cgroups (open /sys/fs/cgroup/memory/memory.limit_in_bytes: no such file or directory)
I160229 18:33:56.700403 server/context.go:229  1 storage engine initialized
I160229 18:33:56.700547 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--
W160229 18:34:23.212798 ts/db.go:96  error writing time series data: failed to send RPC: too many errors encountered (1 of 1 total): server/node.go:620: node 1 stopped
I160229 18:34:23.214183 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestMetricsRecording (0.64s)
=== RUN   TestNewStoreSpec
--- PASS: TestNewStoreSpec (0.02s)
FAIL
FAIL	github.com/cockroachdb/cockroach/server	31.295s
=== RUN   TestMetricsRecorder
--- PASS: TestMetricsRecorder (0.04s)
=== RUN   TestRuntimeStatRecorder
I160229 18:34:02.135051 server/status/runtime.go:137  runtime stats: 4 goroutines, 6.64MiB active, 50000000.00cgo/sec, 986640.00/164440.00 %(u/s)time, 15541.08 %gc (4x)
--- PASS: TestRuntimeStatRecorder (0.00s)
PASS
ok  	github.com/cockroachdb/cockroach/server/status	1.106s
=== RUN   TestSplitOrExpr
--- PASS: TestSplitOrExpr (0.01s)
=== RUN   TestSplitAndExpr

```
Please assign, take a look and update the issue accordingly.

--- a/rpc/codec/rpc_test.go
+++ b/rpc/codec/rpc_test.go
@@ -416,7 +416,6 @@ func benchmarkEchoProtoHTTP(b *testing.B, size int) {
 	go func() {
 		if err := http.Serve(l, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			reqBody, err := ioutil.ReadAll(r.Body)
-			defer r.Body.Close()
 			if err != nil {
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return

--- a/server/authentication_test.go
+++ b/server/authentication_test.go
@@ -131,9 +131,8 @@ func TestSSLEnforcement(t *testing.T) {
 			nodeCertsContext, true, http.StatusOK},
 		{"POST", driver.Endpoint + driver.Execute.String(), sqlForUser(testCertsContext),
 			testCertsContext, true, http.StatusOK},
-		// TODO(bdarnell): https://github.com/cockroachdb/cockroach/issues/4718
-		//{"POST", driver.Endpoint + driver.Execute.String(), sqlForUser(noCertsContext),
-		//	noCertsContext, true, http.StatusUnauthorized},
+		{"POST", driver.Endpoint + driver.Execute.String(), sqlForUser(noCertsContext),
+			noCertsContext, true, http.StatusUnauthorized},
 		{"POST", driver.Endpoint + driver.Execute.String(), sqlForUser(insecureContext),
 			insecureContext, false, -1},
 	}

--- a/sql/server.go
+++ b/sql/server.go
@@ -56,7 +56,6 @@ func MakeServer(ctx *base.Context, executor *Executor) Server {
 // present, in the same format as the request's incoming Content-Type
 // header.
 func (s Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	defer r.Body.Close()
 	method := r.URL.Path
 	if !strings.HasPrefix(method, driver.Endpoint) {
 		http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)

--- a/ts/server.go
+++ b/ts/server.go
@@ -63,7 +63,6 @@ func (s *Server) handleQuery(w http.ResponseWriter, r *http.Request, _ httproute
 
 	// Unmarshal query request.
 	reqBody, err := ioutil.ReadAll(r.Body)
-	defer r.Body.Close()
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return

